react-native/react/reducers/favorite.js

```diff
@@ -0,0 +1,25 @@
+/* @flow */
+
+import * as Constants from '../constants/favorite'
+import type {FavoriteAction} from '../constants/favorite'
+import type {Folder} from '../constants/types/flow-types'
+
+type State = {
+  folders: ?Array<Folder>
+}
+
+const initialState = {
+  folders: null
+}
+
+export default function (state: State = initialState, action: FavoriteAction): State {
+  switch (action.type) {
+    case Constants.favoriteList:
+      if (action.payload) {
```

Comment:
aren't we possibly missing a case here? if !action.payload we'll just return undefined